src/crud/movies.py

```
@@ -0,0 +1,145 @@
+        selectinload(MovieModel.languages),
+    ])
+
+    return movie
+
+async def get_movies(session: AsyncSession, skip: int = 0, limit: int = 10):
+    result = await session.execute(select(MovieModel).offset(skip).limit(limit))
+    movies = result.scalars().all()
+
+    total_result = await session.execute(select(MovieModel))
+    total = len(total_result.scalars().all())
+
+    return movies, total
+
+async def get_movie_by_id(session: AsyncSession, movie_id: int):
+    result = await session.execute(
+        select(MovieModel)
+        .where(MovieModel.id == movie_id)
+        .options(
+            selectinload(MovieModel.country),
+            selectinload(MovieModel.genres),
+            selectinload(MovieModel.actors),
+            selectinload(MovieModel.languages),
+        )
+    )
+    movie = result.scalars().first()
+    return movie
+
+
+async def update_movie(session: AsyncSession, movie_id: int, movie_data: MovieUpdateSchema):
+    result = await session.execute(select(MovieModel).where(MovieModel.id == movie_id))
+    movie = result.scalars().first()
+    if not movie:
+        return None
+
+    for field, value in movie_data.dict(exclude_unset=True).items():
+        if field == "genre_ids":
+            genres = await session.execute(select(GenreModel).where(GenreModel.id.in_(value)))
+            movie.genres = genres.scalars().all()
+        elif field == "actor_ids":
+            actors = await session.execute(select(ActorModel).where(ActorModel.id.in_(value)))
+            movie.actors = actors.scalars().all()
+        elif field == "language_ids":
+            languages = await session.execute(select(LanguageModel).where(LanguageModel.id.in_(value)))
+            movie.languages = languages.scalars().all()
+        elif field == "country_id":
+            country = await session.get(CountryModel, value)
+            movie.country = country
+        else:
+            setattr(movie, field, value)

```

Potential issue: The update logic expects fields like 'genre_ids', 'actor_ids', 'language_ids', and 'country_id' in the MovieUpdateSchema. Please ensure that your schema defines these fields with these exact names and types (list of IDs for genres/actors/languages, single ID for country). If the schema uses different names or expects nested objects, this update logic will not work as intended. Double-check the schema to avoid runtime errors or silent failures.